homeassistant/components/mqtt/device_trigger.py

```diff
@@ -275,8 +275,8 @@ async def discovery_update(payload):
     async_dispatcher_send(hass, MQTT_DISCOVERY_DONE.format(discovery_hash), None)
 
 
-async def async_device_removed(hass: HomeAssistant, device_id: str):
-    """Handle the removal of a device."""
+async def async_removed_from_device(hass: HomeAssistant, device_id: str):
```

Comment:
Is there no common logic that detaches the device trigger of a removed device? This sounds like something all integrations with device triggers may need. Integrations may need integration specific handling too but detaching the trigger should be general, right?

Reply:
Yes, it might be useful to add a device trigger function to clean up stale triggers etc. when a device (or an entity which provides device triggers) is removed. 👍 
That will be quite the effort though, we need to do that in a group of separate PRs.